src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -482,6 +482,11 @@ public ProcessStartInfo StartInfo
             {
                 if (_startInfo == null)
                 {
+                    if (Associated && Process.GetCurrentProcessId() != Id)
```

Comment:
Why do you allows devs to get the StartInfo for the current process? It also contains bogus information; environment variables is the only thing that returns something, and even then it is snapshotted and never changed once you've read it.